### PR TITLE
Abstract concurrency-related operations as ConcurrencyStat interface and polish StatNode

### PR DESCRIPTION
**Pull request by louyuting**
### Special notes for reviews

### CODE REVIEW

File: core/stat/base_node.go [hunk 1]
```diff
@@ -66,23 +66,18 @@ func (n *BaseStatNode) MinRT() float64 {
 	return float64(n.metric.MinRT())
 }
 
-func (n *BaseStatNode) CurrentGoroutineNum() int32 {
+func (n *BaseStatNode) CurrentConcurrency() int32 {
```
Comment (luckyxiaoqiang) [inline core/stat/base_node.go:69]: How about renaming `goroutineNum` to `concurrency`?

Comment (louyuting) [reply core/stat/base_node.go:69]: make sense

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**